src/connection_string.ts

```diff
@@ -310,6 +310,17 @@ export function parseOptions(
     );
   }
 
+  const uriMechanismProperties = urlOptions.get('authMechanismProperties');
+  if (uriMechanismProperties) {
+    for (const property of uriMechanismProperties) {
+      if (property.includes('ALLOWED_HOSTS:')) {
```

Comment:
```suggestion
      if (/(^|,)ALLOWED_HOSTS:/.test(property)) {
```

so that you don’t accidentally run into the (of course very unlikely) case that `ALLOWED_HOSTS:` is part of the value of another auth mechanism property?

Reply:
Update to suggestion.